src/@chakra-ui/gatsby-plugin/components/components.utils.ts

```diff
@@ -70,3 +72,62 @@ export function defineMergeStyles(
 ): Record<string, SystemStyleObject> {
   return merge(defaultTheming, ...styleObjs)
 }
+
+export const _notDisabledReadOnly =
+  "&:not([data-disabled], [disabled], [data-readonly])"
+
+const $inputTriggerDisableColor = cssVar("input-trigger-disable-color")
+
+export const commonInputTriggerStyles = {
+  commonControlProps: {
+    border: "1px",
+    borderColor: "body.medium",
+    outline: "3px solid",
+    outlineColor: "transparent",
+    _checked: {
+      color: "background.base",
+      bg: "primary.base",
+      borderColor: "primary.base",
+    },
+    _focusVisible: {
+      borderColor: "primary.highContrast",
+      outlineColor: "primary.hover",
+      outlineOffset: "2px",
+      boxShadow: "none",
+    },
+    _disabled: {
+      bg: $inputTriggerDisableColor.reference,
+      borderColor: $inputTriggerDisableColor.reference,
+      opacity: 1,
+      _checked: {
+        bg: $inputTriggerDisableColor.reference,
+        borderColor: $inputTriggerDisableColor.reference,
+      },
+    },
+    [_notDisabledReadOnly]: {
+      // Hovering over the label triggers the style for the control
+      "*:hover > &": {
+        bg: "body.light",
+        borderColor: "primary.highContrast",
+      },
+    },
+    _invalid: {
+      // TODO: Investigate inconsistency in prop rendering order (possible Chakra bug)
+      // border: "2px",
```

Comment:
For whatever reason, using `border` and `borderColor` here works fine for Checkbox, but rendered in reverse order for Radio, cause the `borderColor` styling to be overriden in the CSS by the `border`.
![image](https://github.com/ethereum/ethereum-org-website/assets/65234762/45855ab1-591b-468b-9bb1-6d5e119e519a)

Now this is probably due to using the default theming which includes an `invalid` styling, but I think a refactor in how I am applying these common props might fix it (send  these props to the checkbox and then import the checkbox styles to the radio and switch). Otherwise, this might be a compiling bug.

Either way, I think the reorganizing and addressing the prop issue should be saved for a separate PR to get this PR over the finish line... unless you would like for it to be resolved here.

Reply:
Ok, thanks @TylerAPfledderer.